### ISSUE TITLE
Abstraction for `NamingPolicy` and temp dir

### ISSUE THREAD

**Issue by olenagerasimova** [artipie/rpm-adapter#315]
Let's create abstraction (or modify existing code, `NamingPolicy` for example) for `NamingPolicy` and temp dir. The main goals here are
a) not to pass `tmp` dir into `Repository` ctor, it's required only in `Repository#save()` method
b) reduce parameters amount in `MetadataFile#save()`

Do not forget about test.

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@661823d7-f32c-4fcc-bf9e-6e35e0472a13, thread: PQ-CT2E6TK9B
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 2 error(s) in XML document: -1:-1: cvc-pattern-valid: Value 'gh:Week 28' is not facet-valid with respect to pattern '[a-z]{2}:[A-Z0-9a-z.\-#/]+' for type 'milestoneID'.;-1:-1: cvc-complex-type.2.2: Element 'successor' must have no element [children], and the value must be valid.
```

[0.53.15](https://github.com/zerocracy/farm): CID: [79b5eba6-203f-4a9d-b795-de240c3df272](https://www.0crat.com/footprint/CT2E6TK9B/79b5eba6-203f-4a9d-b795-de240c3df272), Type: "Job milestoned"

**Comment by 0crat** [comment]
The job [#315](https://github.com/artipie/rpm-adapter/issues/315) assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/a5bbecac-c703-42ba-bff3-798f8cd09e56); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by olenagerasimova** [comment]
Closed via #322

**Comment by 0crat** [comment]
Job was finished in 4 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))
